File: app/src/main/java/cn/com/szw/lib/myframework/utils/DialogType104.java
package cn.com.szw.lib.myframework.utils;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.common.controls.dialog.DialogBase;
import com.common.controls.dialog.DialogUtil;

/**
 * 类型104 Dialog
 * 有标题，有自定义布局内容，两个按钮
 */

public class DialogType104 extends DialogBase {
    private FrameLayout mContentContainer;
    private TextView mCancelTv;
    private TextView mOkTv;


    private View mMiddleDivider;
    private OnBeforeDismissListener onBeforeDismiss;

    public void setOnBeforeDismiss(OnBeforeDismissListener onBeforeDismiss) {
        this.onBeforeDismiss = onBeforeDismiss;
    }

    public DialogType104(Context context) {
        super(context);
    }

    @Override
    public void dismiss() {
        if (onBeforeDismiss != null)
            if (onBeforeDismiss.onBeforeDismiss()) {
                if (mDialog != null) mDialog.dismiss();
            }
    }

    public interface OnBeforeDismissListener {
        boolean onBeforeDismiss();
    }

    @Override
    public void initDialog() {
        View rootView = LayoutInflater.from(mContext).inflate(com.common.alertpop.R.layout.common_dialog_layout_type104, null);
        mTitleTv = (TextView) rootView.findViewById(com.common.alertpop.R.id.common_dialog_title_text);
        mContentContainer = (FrameLayout) rootView.findViewById(com.common.alertpop.R.id.common_dialog_content_container);
        mCancelTv = (TextView) rootView.findViewById(com.common.alertpop.R.id.common_dialog_cancel_btn);
        mOkTv = (TextView) rootView.findViewById(com.common.alertpop.R.id.common_dialog_ok_btn);
        mMiddleDivider = rootView.findViewById(com.common.alertpop.R.id.common_dialog_btn_middle_divider);
        createDialog(rootView);
    }

    @Override
    public void setContentView(int contentLayoutId) {
        LayoutInflater.from(mContext).inflate(contentLayoutId, mContentContainer, true);
    }

    @Override
    public void setContentView(View contentView) {
        mContentContainer.addView(contentView);
    }

    @Override
    public void setContentView(View contentView, ViewGroup.LayoutParams params) {
        mContentContainer.addView(contentView, params);
    }

    @Override
    public void setCancelBtn(int textId, View.OnClickListener cancelOnClickListener) {
        mCancelTv.setText(textId);
        setOnCancelClickListener(cancelOnClickListener);
    }

    @Override
    public void setCancelBtn(CharSequence text, View.OnClickListener cancelOnClickListener) {
        mCancelTv.setText(text);
        setOnCancelClickListener(cancelOnClickListener);
    }

    @Override
    public void setOkBtn(int textId, View.OnClickListener okOnClickListener) {
        mOkTv.setText(textId);
        setOnOkClickListener(okOnClickListener);
    }

    @Override
    public void setOkBtn(CharSequence text, View.OnClickListener okOnClickListener) {
        mOkTv.setText(text);
        setOnOkClickListener(okOnClickListener);
    }

    @Override
    public void setOkBtnStyleType(int okBtnStyleType) {
        mOkTv.setBackgroundResource(DialogUtil.getOkBtnBgResId(okBtnStyleType));
        mOkTv.setTextColor(DialogUtil.getOkBtnTextColorValue(okBtnStyleType, mOkTv.getContext()));
        mMiddleDivider.setVisibility(DialogUtil.isBtnMiddleDividerVisible(okBtnStyleType) ? View.VISIBLE : View.GONE);
    }

    @Override
    public void bindAllListeners() {
        mCancelTv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DialogType104.this.onCancelClick(v);
            }
        });
        mOkTv.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                DialogType104.this.onOkClick(v);
            }
        });
    }

}

File: Flibrary/src/main/java/com/szw/framelibrary/utils/DialogType104.kt
package com.szw.framelibrary.utils

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import android.widget.TextView

import com.common.controls.dialog.DialogBase
import com.common.controls.dialog.DialogUtil

/**
 * 类型104 Dialog
 * 有标题，有自定义布局内容，两个按钮
 */

class DialogType104(context: Context) : DialogBase(context) {
    private var mContentContainer: FrameLayout? = null
    private var mCancelTv: TextView? = null
    private var mOkTv: TextView? = null


    private var mMiddleDivider: View? = null
    private var onBeforeDismiss: OnBeforeDismissListener? = null

    fun setOnBeforeDismiss(onBeforeDismiss: OnBeforeDismissListener) {
        this.onBeforeDismiss = onBeforeDismiss
    }

    override fun dismiss() {
        if (onBeforeDismiss != null)
            if (onBeforeDismiss?.onBeforeDismiss()==true) {
                if (mDialog != null) mDialog.dismiss()
            }
    }

    interface OnBeforeDismissListener {
        fun onBeforeDismiss(): Boolean
    }

    override fun initDialog() {
        val rootView = LayoutInflater.from(mContext).inflate(com.common.alertpop.R.layout.common_dialog_layout_type104, null)
        mTitleTv = rootView.findViewById<View>(com.common.alertpop.R.id.common_dialog_title_text) as TextView
        mContentContainer = rootView.findViewById<View>(com.common.alertpop.R.id.common_dialog_content_container) as FrameLayout
        mCancelTv = rootView.findViewById<View>(com.common.alertpop.R.id.common_dialog_cancel_btn) as TextView
        mOkTv = rootView.findViewById<View>(com.common.alertpop.R.id.common_dialog_ok_btn) as TextView
        mMiddleDivider = rootView.findViewById(com.common.alertpop.R.id.common_dialog_btn_middle_divider)
        createDialog(rootView)
    }

    override fun setContentView(contentLayoutId: Int) {
        LayoutInflater.from(mContext).inflate(contentLayoutId, mContentContainer, true)
    }

    override fun setContentView(contentView: View?) {
        mContentContainer?.addView(contentView)
    }

    override fun setContentView(contentView: View?, params: ViewGroup.LayoutParams?) {
        mContentContainer?.addView(contentView, params)
    }

    override fun setCancelBtn(textId: Int, cancelOnClickListener: View.OnClickListener?) {
        mCancelTv?.setText(textId)
        setOnCancelClickListener(cancelOnClickListener)
    }

    override fun setCancelBtn(text: CharSequence?, cancelOnClickListener: View.OnClickListener?) {
        mCancelTv?.text = text
        setOnCancelClickListener(cancelOnClickListener)
    }

    override fun setOkBtn(textId: Int, okOnClickListener: View.OnClickListener?) {
        mOkTv?.setText(textId)
        setOnOkClickListener(okOnClickListener)
    }

    override fun setOkBtn(text: CharSequence?, okOnClickListener: View.OnClickListener?) {
        mOkTv?.text = text
        setOnOkClickListener(okOnClickListener)
    }

    override fun setOkBtnStyleType(okBtnStyleType: Int) {
        mOkTv?.setBackgroundResource(DialogUtil.getOkBtnBgResId(okBtnStyleType))
        mOkTv?.setTextColor(DialogUtil.getOkBtnTextColorValue(okBtnStyleType, mOkTv?.context))
        mMiddleDivider?.visibility = if (DialogUtil.isBtnMiddleDividerVisible(okBtnStyleType)) View.VISIBLE else View.GONE
    }

    override fun bindAllListeners() {
        mCancelTv?.setOnClickListener { v -> this@DialogType104.onCancelClick(v) }
        mOkTv?.setOnClickListener { v -> this@DialogType104.onOkClick(v) }
    }

}
